source/common/http/http1/codec_impl.cc

```diff
@@ -575,7 +594,10 @@ StreamEncoder& ClientConnectionImpl::newStream(StreamDecoder& response_decoder)
   if (resetStreamCalled()) {
     throw CodecClientException("cannot create new streams after calling reset");
   }
-
+  // read-enable any underlying connection disabled by flow control on the prior request.
+  while (!connection_.readEnabled()) {
```

Comment:
The while loop here makes me a bit uneasy. Aren't we undoing read disables from someone else that they might enable later and thus hit assert or some other problem? If this is safe can you comment why?